File: frontend/src/types/student.ts
export type Parentesco = 
  | "Pai" | "Mãe" | "Avô" | "Avó" | "Tio" | "Tia" | "Bisavô" | "Bisavó" | "Cunhado(a)" | "Família Acolhedora" | "Curador(a)"
  | "Primo" | "Prima" | "Tutor(a) Legal" |"Irmão" | "Irmã" | "Madrasta" | "Padrasto" | "Outro";
export type Sexo = "Masculino" | "Feminino" | "Outro";
export type RacaCor = "Branca" | "Preta" | "Parda" | "Amarela" | "Indígena" | "Não Declarado";
export type EstadoCivil = "Solteiro(a)" | "Casado(a)" | "Divorciado(a)" | "Viúvo(a)" | "União Estável";
export type Uf = 'AC' | 'AL' | 'AP' | 'AM' | 'BA' | 'CE' | 'DF' | 'ES' | 'GO' | 'MA' | 'MT' | 'MS' | 'MG' | 'PA' | 'PB' | 'PR' | 'PE' | 'PI' | 'RJ' | 'RN' | 'RS' | 'RO' | 'RR' | 'SC' | 'SP' | 'SE' | 'TO';

export interface ResponsavelLegal {
  nome: string;
  dataNascimento: string;
  rg: string;
  cpf: string;
  celular: string;
  operadora: string;
  whatsapp: string;
  fixo: string;
  parentesco: Parentesco | "";
}

export interface MembroFamiliar {
  nome: string;
  parentesco: Parentesco | "";
  profissao: string;
  renda: string;
}

export interface PessoaAutorizada {
  nome: string;
  documento: string;
  parentesco: Parentesco | "";
  telefone: string;
}

export type AutorizacaoSaida =
  | ""
  | "sim"
  | "nao"
  | "somente-com-responsavel";

export interface Student {
  id: string;
  createdAt: string;
  nomeCompleto: string;
  fotoCrianca: string;
  locomocao: string;
  locomocaoAcompanhante: string;
  origemEncaminhamento: string;
  dataNascimento: string;
  idade: number | null;
  naturalidade: string;
  racaCor: RacaCor | "";
  sexo: Sexo | "";
  rg: string;
  cpf: string;
  nis: string;
  certidaoTermo: string;
  certidaoFolha: string;
  certidaoLivro: string;
  enderecoCep: string;
  enderecoLogradouro: string;
  enderecoNumero: string;
  enderecoComplemento: string;
  enderecoBairro: string;
  enderecoCidade: string;
  enderecoUf: Uf | "";
  nomePai: string;
  nomeMae: string;
  crasReferencia: string;
  responsaveisLegais: ResponsavelLegal[];
  membrosFamiliares: MembroFamiliar[];
  estadoCivilPais: EstadoCivil | "";
  contatoConjugeNome: string;
  contatoConjugeTelefone: string;
  contatoConjugeFrequencia: string;
  tipoDomicilio: string;
  rendaFamiliar: string;
  faixaRenda: string;
  beneficios: string[];
  beneficioOutros: string;
  ativo: boolean;
  unidade: string;
  escolaNome: string;
  escolaSerie: string;
  escolaAno: string;
  escolaProfessor: string;
  escolaPeriodo: string;
  historicoEscolar: string;
  evasaoEscolar: boolean;
  evasaoEscolarMotivo: string;
  evasaoEscolarTempo: string;
  ubsReferencia: string;
  locaisAtendimento: string[];
  temProblemaSaude: boolean;
  problemaSaudeDescricao: string;
  temRestricoes: boolean;
  restricoesDescricao: string;
  usaMedicamentos: boolean;
  medicamentosDescricao: string;
  temAlergias: boolean;
  alergiasDescricao: string;
  acompanhamentos: string;
  temDeficiencia: boolean;
  deficienciaDescricao: string;
  temBronquite: boolean;
  temFaltaAr: boolean;
  acompanhamentoOdontologico: boolean;
  acompanhamentoOdontologicoLocal: string;
  acompanhamentoOdontologicoTempo: string;
  tratamentoOftalmologico: boolean;
  tratamentoOftalmologicoLocal: string;
  usaOculos: boolean;
  usaLentes: boolean;
  restricaoFisica: boolean;
  restricaoFisicaDescricao: string;
  permaneceSozinhaEmCasa: boolean;
  temSupervisao: boolean;
  supervisaoDescricao: string;
  frequenciaInteracao: string;
  interacaoSocial: string[];
  locaisLazer: string[];
  atividadesExtras: string;
  atividadesExtrasLista: string[];
  cronogramaAtividades: string[];
  servicosUtilizados: string[];
  situacaoPrioritaria: boolean;
  observacoesGerais: string;
  termoResponsabilidade: boolean;
  autorizacaoImagem: boolean;
  autorizacaoSaida: AutorizacaoSaida;
  pessoasAutorizadas: PessoaAutorizada[];
}

export const emptyStudent: Omit<Student, "id"> = {
  createdAt: "",
  nomeCompleto: "",
  fotoCrianca: "",
  locomocao: "",
  locomocaoAcompanhante: "",
  origemEncaminhamento: "",
  dataNascimento: "",
  idade: null,
  naturalidade: "",
  racaCor: "",
  sexo: "",
  rg: "",
  cpf: "",
  nis: "",
  certidaoTermo: "",
  certidaoFolha: "",
  certidaoLivro: "",
  enderecoCep: "",
  enderecoLogradouro: "",
  enderecoNumero: "",
  enderecoComplemento: "",
  enderecoBairro: "",
  enderecoCidade: "",
  enderecoUf: "",
  nomePai: "",
  nomeMae: "",
  crasReferencia: "",
  responsaveisLegais: [
    {
      nome: "",
      dataNascimento: "",
      rg: "",
      cpf: "",
      celular: "",
      operadora: "",
      whatsapp: "",
      fixo: "",
      parentesco: "",
    },
  ],
  membrosFamiliares: [],
  estadoCivilPais: "",
  contatoConjugeNome: "",
  contatoConjugeTelefone: "",
  contatoConjugeFrequencia: "",
  tipoDomicilio: "",
  rendaFamiliar: "",
  faixaRenda: "",
  beneficios: [],
  beneficioOutros: "",
  ativo: true,
  unidade: "",
  escolaNome: "",
  escolaSerie: "",
  escolaAno: "",
  escolaProfessor: "",
  escolaPeriodo: "",
  historicoEscolar: "",
  evasaoEscolar: false,
  evasaoEscolarMotivo: "",
  evasaoEscolarTempo: "",
  ubsReferencia: "",
  locaisAtendimento: [],
  temProblemaSaude: false,
  problemaSaudeDescricao: "",
  temRestricoes: false,
  restricoesDescricao: "",
  usaMedicamentos: false,
  medicamentosDescricao: "",
  temAlergias: false,
  alergiasDescricao: "",
  acompanhamentos: "",
  temDeficiencia: false,
  deficienciaDescricao: "",
  temBronquite: false,
  temFaltaAr: false,
  acompanhamentoOdontologico: false,
  acompanhamentoOdontologicoLocal: "",
  acompanhamentoOdontologicoTempo: "",
  tratamentoOftalmologico: false,
  tratamentoOftalmologicoLocal: "",
  usaOculos: false,
  usaLentes: false,
  restricaoFisica: false,
  restricaoFisicaDescricao: "",
  permaneceSozinhaEmCasa: false,
  temSupervisao: false,
  supervisaoDescricao: "",
  frequenciaInteracao: "",
  interacaoSocial: [],
  locaisLazer: [],
  atividadesExtras: "",
  atividadesExtrasLista: [],
  cronogramaAtividades: [],
  servicosUtilizados: [],
  situacaoPrioritaria: false,
  observacoesGerais: "",
  termoResponsabilidade: false,
  autorizacaoImagem: false,
  autorizacaoSaida: "",
  pessoasAutorizadas: [],
};

File: backend/src/models/student.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StudentListItem {
  pub id: String,
  pub nome_completo: String,
  pub idade: Option<i32>,
  pub nome_mae: Option<String>,
  pub escola_nome: Option<String>,
  pub sexo: Option<String>,
  pub cpf: Option<String>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StudentAuditEvent {
  pub id: String,
  pub student_id: String,
  pub student_name: String,
  pub action: String,
  pub at: String,
  pub by_email: String,
  pub changed_fields: Option<Vec<String>>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ResponsavelLegal {
  pub nome: String,
  pub data_nascimento: String,
  pub rg: String,
  pub cpf: String,
  pub celular: String,
  pub operadora: String,
  pub whatsapp: String,
  pub fixo: String,
  pub parentesco: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct MembroFamiliar {
  pub nome: String,
  pub parentesco: String,
  pub profissao: String,
  pub renda: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct PessoaAutorizada {
  pub nome: String,
  pub documento: String,
  pub parentesco: String,
  pub telefone: String,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct StudentDraft {
  pub nome_completo: String,
  pub data_nascimento: String,
  pub idade: Option<i32>,
  pub naturalidade: String,
  pub raca_cor: String,
  pub sexo: String,
  pub rg: String,
  pub cpf: String,
  pub nis: String,
  pub certidao_termo: String,
  pub certidao_folha: String,
  pub certidao_livro: String,
  pub endereco_cep: String,
  pub endereco_logradouro: String,
  pub endereco_numero: String,
  pub endereco_complemento: String,
  pub endereco_bairro: String,
  pub endereco_cidade: String,
  pub endereco_uf: String,
  pub nome_pai: String,
  pub nome_mae: String,
  pub cras_referencia: String,
  pub responsaveis_legais: Vec<ResponsavelLegal>,
  pub membros_familiares: Vec<MembroFamiliar>,
  pub estado_civil_pais: String,
  pub contato_conjuge_nome: String,
  pub contato_conjuge_telefone: String,
  pub tipo_domicilio: String,
  pub renda_familiar: String,
  pub beneficios: Vec<String>,
  pub escola_nome: String,
  pub escola_serie: String,
  pub escola_ano: String,
  pub escola_professor: String,
  pub escola_periodo: String,
  pub historico_escolar: String,
  pub ubs_referencia: String,
  pub tem_problema_saude: bool,
  pub problema_saude_descricao: String,
  pub tem_restricoes: bool,
  pub restricoes_descricao: String,
  pub usa_medicamentos: bool,
  pub medicamentos_descricao: String,
  pub tem_alergias: bool,
  pub alergias_descricao: String,
  pub acompanhamentos: String,
  pub tem_deficiencia: bool,
  pub deficiencia_descricao: String,
  pub tem_supervisao: bool,
  pub supervisao_descricao: String,
  pub interacao_social: Vec<String>,
  pub locais_lazer: Vec<String>,
  pub atividades_extras: String,
  pub servicos_utilizados: Vec<String>,
  pub termo_responsabilidade: bool,
  pub autorizacao_imagem: bool,
  pub autorizacao_saida: String,
  pub pessoas_autorizadas: Vec<PessoaAutorizada>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct Student {
  pub id: String,
  #[serde(flatten)]
  pub data: StudentDraft,
}

